From 5c6e712c3ffbc178a8410cb9eb58815371a6e6a1 Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Mon, 1 May 2023 08:34:01 +0200
Subject: [PATCH 02/12] Replace gapps with MindTheGapps

---
 generate.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/generate.sh b/generate.sh
index 9c20eb5..d016d07 100644
--- a/generate.sh
+++ b/generate.sh
@@ -26,8 +26,8 @@ for part in a ab;do
 		optional_base=""
 				if [ "$apps" == "gapps" ];then
 					apps_suffix="g"
-					apps_script='$(call inherit-product, device/phh/treble/gapps.mk)'
-					apps_name="with GApps"
+					apps_script='$(call inherit-product, vendor/gapps/'$baseArch'/'$baseArch'-vendor.mk)'
+					apps_name="with MindTheGapps"
 				fi
 				if [ "$apps" == "gapps-go" ];then
 					apps_suffix="o"
-- 
2.33.0.windows.2

