From c15e7ffc467dc9fbc456df5ff784c6a075be7dcc Mon Sep 17 00:00:00 2001
From: utkustnr <utkuustuner@proton.me>
Date: Mon, 14 Nov 2022 20:51:04 +0300
Subject: [PATCH] [1/2] Replace backup provider with seedvault

---
 base.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/base.mk b/base.mk
index ce2f218..59ba167 100644
--- a/base.mk
+++ b/base.mk
@@ -248,3 +248,6 @@ include build/make/target/product/gsi_release.mk
 # Protect deskclock from power save
 PRODUCT_COPY_FILES += \
 	device/phh/treble/files/com.android.deskclock_whitelist.xml:system/etc/sysconfig/com.android.deskclock_whitelist.xml
+
+# Replace backup provider with seedvault
+PRODUCT_PACKAGES += Seedvault
\ No newline at end of file
-- 
2.34.1

