From c21743430354b883454bfc207501ac0073a2e30e Mon Sep 17 00:00:00 2001
From: utkustnr <utkuustuner@proton.me>
Date: Mon, 14 Nov 2022 20:51:04 +0300
Subject: [PATCH 7/7] Replace backup provider with seedvault

---
 base.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/base.mk b/base.mk
index a83d370..71b1ce9 100644
--- a/base.mk
+++ b/base.mk
@@ -264,3 +264,6 @@ include build/make/target/product/gsi_release.mk
 # Protect deskclock from power save
 PRODUCT_COPY_FILES += \
 	device/phh/treble/files/com.android.deskclock_whitelist.xml:system/etc/sysconfig/com.android.deskclock_whitelist.xml
+
+# Replace backup provider with seedvault
+PRODUCT_PACKAGES += Seedvault
\ No newline at end of file
-- 
2.25.1

