From 4222d40b4f1df727097564b0e58712db246731f8 Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Wed, 21 Dec 2022 09:14:56 +0100
Subject: [PATCH 1/2] Change product_name to leaos

---
 generate.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/generate.sh b/generate.sh
index e83345d..e692737 100644
--- a/generate.sh
+++ b/generate.sh
@@ -54,7 +54,7 @@ for part in a ab;do
 				su_suffix='N'
 				if [ "$su" == "yes" ];then
 					su_suffix='S'
-					extra_packages+=' phh-su SuperUser'
+					extra_packages+=' phh-su me.phh.superuser'
 				fi
 
 				part_suffix='a'
@@ -85,7 +85,7 @@ $optional_base
 $apps_script
 $rom_script
 
-PRODUCT_NAME := $target
+PRODUCT_NAME := LeaOS
 PRODUCT_DEVICE := tdgsi_${arch}_$part
 PRODUCT_BRAND := google
 PRODUCT_SYSTEM_BRAND := google
-- 
2.25.1

