From 649a37390f5d8aa1c4fa723103704c3ad4aaa2f4 Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Fri, 31 Mar 2023 20:38:39 +0200
Subject: [PATCH] Fix Huawei emui9 denied - update one

---
 sepolicy/huawei.te | 31 ++++++++++++++++++++++++++-----
 1 file changed, 26 insertions(+), 5 deletions(-)

diff --git a/sepolicy/huawei.te b/sepolicy/huawei.te
index d5c89a7..61b20e3 100644
--- a/sepolicy/huawei.te
+++ b/sepolicy/huawei.te
@@ -51,8 +51,9 @@ allow vendor_init splash2_data_file:filesystem { getattr };
 allow installd splash2_data_file:filesystem { quotaget };
 allow rild splash2_data_file:dir rw_dir_perms;
 allow rild splash2_data_file:file rw_file_perms;
-allow kernel splash2_data_file:dir { rw_dir_perms create setattr };
-allow kernel splash2_data_file:file rw_file_perms;
+allow kernel splash2_data_file:dir { rw_dir_perms create setattr rmdir };
+allow kernel splash2_data_file:file { rw_file_perms create unlink };
+
 
 
 # Fix CPU gouvernor (rules are in xml file)
@@ -78,7 +79,7 @@ type sysfs_led, domain;
 allow init sysfs_led:file { setattr };
 
 
-#allow crash_dump bluetooth_data_file:file r_dir_perms;
+#allow crash_dump bluetooth_data_file:file rw_file_perms;
 #allow crash_dump resourcecache_data_file:file { read open };
 
 allow vold sys_block_mmcblk0:file { write };
@@ -131,6 +132,19 @@ allow system_server vendor_file:file r_file_perms;
 allow kernel sysfs_devices_system_cpu:file rw_file_perms;
 allow network_stack fs_bpf:file rw_file_perms;
 
+
+type hi110x_cust_data_file, file_type; 
+type mnt_modem_file, file_type;  
+
+allow wpa_hisi hi110x_cust_data_file:dir r_dir_perms;
+allow gmscore_app mnt_modem_file:dir r_dir_perms;
+
+allow kernel anr_data_file:dir r_dir_perms;
+allow bootanim system_data_file:dir r_dir_perms;
+
+allow priv_app gmscore_app:file { read };
+
+
 # OTPERASE = 0x4d19
 allowxperm oeminfo_nvm device:blk_file ioctl {
   0x4d19
@@ -147,17 +161,21 @@ allowxperm phhsu_daemon device:blk_file ioctl {
   BLKDISCARDZEROES
 };
 
-
 allow shell self:netlink_kobject_uevent_socket { create setopt getopt bind read };
 allow init self:netlink_kobject_uevent_socket { create setopt getopt bind read };
 
+# DRM
+type hal_drm_widevine, domain;
+allow hal_drm_widevine shell_data_file:dir { search };
+allow hal_drm_widevine shell_test_data_file:dir { search };
+
 
 # Authorize a lot of process to read default, config and system prop
 type hal_health_default, domain;
 type hal_camera_default, domain;
 type hal_audio_default, domain;
 type hal_keymaster_default, domain;
-type hal_drm_widevine, domain;
+
 
 get_prop(rild, config_prop)
 get_prop(rild, system_prop)
@@ -190,3 +208,6 @@ get_prop(atcmdserver, default_prop)
 get_prop(hisecd, default_prop)
 get_prop(mediaserver, vendor_default_prop)
 get_prop(storage_info, default_prop)
+get_prop(audioserver, vendor_default_prop)
+
+
-- 
2.25.1

